src/c++/perf_analyzer/load_manager.cc

```diff
@@ -143,6 +143,42 @@ LoadManager::ResetIdleTime()
   }
 }
 
+double
+LoadManager::GetSendRequestRate()
+{
+  double send_request_rate{0.0};
+  std::chrono::steady_clock::time_point now{std::chrono::steady_clock::now()};
+  size_t num_requests_sent{0};
+  std::chrono::steady_clock::time_point earliest_send{
+      std::chrono::steady_clock::time_point::max()};
+  std::chrono::steady_clock::time_point latest_send{
+      std::chrono::steady_clock::time_point::min()};
+
+  for (auto& thread_stat : threads_stat_) {
+    std::lock_guard<std::mutex> lock(thread_stat->mu_);
+    if (thread_stat->request_send_times_.empty() == false &&
+        thread_stat->request_send_times_.front() < earliest_send) {
+      earliest_send = thread_stat->request_send_times_.front();
```

Comment:
I don't like this computation of window. The load manager shouldn't understand windows, and this method of determining the window isn't fully accurate. Why don't we just count the requests in an integer (instead of a vector of timestamps), and then let inference_profiler divide by the window size?

Reply:
Changed to your new design 👍